pkg/channel/consolidated/reconciler/controller/kafkachannel.go

```diff
@@ -477,7 +492,20 @@ func (r *Reconciler) reconcileChannelService(ctx context.Context, dispatcherName
 	return svc, nil
 }
 
-func (r *Reconciler) createClient(ctx context.Context) (sarama.ClusterAdmin, error) {
+func (r *Reconciler) createClients(ctx context.Context) (sarama.Client, sarama.ClusterAdmin, error) {
+	kafkaClient := r.kafkaClient
```

Comment:
you're not setting 
`r.kafkaClient` anywhere, is that intentional? wouldn't that mean you're creating a new client with every reconcile loop?

Reply:
I refactored the code that creates the admin client and created the normal client there.

For clusterAdmin, there's this ticket: https://github.com/Shopify/sarama/issues/1162. I see that ticket didn't really get any resolution.

I haven't tried reusing the regular kafkaClient. I can try that, once I made the failing tests working.

Reply:
pretty bad sarama bug :disappointed:

Reply:
Can we add a comment that this thing returns both clients? Normal and ClusterAdmin?

`func (r *Reconciler) createSaramaClients` instead?  not just client?

Reply:
Do we really always need _both_?  See Ahmed's comment with stashing and defering the "regular" client.

Just wondering

Reply:
one alternative can be that we have two functions, one for each.

Reply:
We now have 2 functions. Code looks much better, thanks for the suggestions.

About stashing into `r.kafkaClient` -> I will try this, one I see tests are passing again.

Reply:
Ok, when I stash the client to `r.kafkaClient` and reuse it, things break. I have no desire to debug why.

See my comment here: https://github.com/knative-sandbox/eventing-kafka/pull/886#issuecomment-932092014

You can have a look at the commit list here, https://github.com/knative-sandbox/eventing-kafka/pull/886/commits, and see the job history for each commit there (not always consistent, especially when there's new code pushed while the jobs were running).

Reply:
> . I have no desire to debug why.

that is OK w/ me. That said - mind creating an issue we are having to track this?

Reply:
https://github.com/knative-sandbox/eventing-kafka/issues/912